src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -496,6 +535,82 @@ internal void ProcessDto(ModelBindingContext bindingContext, ComplexModelDto dto
             }
         }
 
+        // Neither [DefaultValue] nor [Required] is relevant for a non-settable collection.
+        private void AddToProperty(
+            ModelBindingContext bindingContext,
+            ModelExplorer modelExplorer,
```

Comment:
Note that this code is also using model explorer - that's fine from the POV of using `GetMetadataForProperty`, but it will also use the runtime type if you try to get it's properties. 

From the usage it doesn't seem to cause problems, but we might want to clobber it if we refactor this code.